### PR TITLE
add homework for week2

### PR DESCRIPTION
**Pull request by zealotnt**
Hi anh Hiến,

I'd like to submit my homework for week 2.
Sorry that I'm too busy so that I can't afford time to finish it sooner.

Cheers,
Tam

### CODE REVIEW

File: homeworks/week2/Tam-Tran.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    uniqChars = {}
+    ans = []
+    for ch in inputVal:
+        if ch not in uniqChars:
+            ans.append(ch)
+            uniqChars[ch] = True
+    return ''.join(ans)
+
+testCases = [
```
Comment (ubriela) [inline homeworks/week2/Tam-Tran.py:23]: Nice test cases!

File: homeworks/week2/Tam-Tran.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# Time complexity: O(n):
+# - we iterate through the input string/array once
+# - look up to a checked hash once
+# - and append to the result array (the size of array is capped at string encoding size), we can improve this to use a link list
+#   which has a O(1) for add an item to it
+# Space complexity: O(1): the data structure we create is:
+# - a result string (it is capped at string encoding standard size)
+# - a hashmap collection of visited characters (capped at string encodinng size as well)
+def removeDuplicateCharacters(inputVal):
```
Comment (ubriela) [inline homeworks/week2/Tam-Tran.py:11]: nit: these camelCase code style is Java.

For python, I believe a more common naming would be
```
def remove_duplicate_characters(input_val):
```

**Review by ubriela — COMMENTED** [review]
If this is an interview output for an entry level, I'd say Hire.

File: homeworks/week2/Tam-Tran.py (hunk 2 above)
Comment (zealotnt) [reply homeworks/week2/Tam-Tran.py:11]: Yeah thanks for your useful feedback, havent code python nearly a year, should I correct it before PR can be merged ?

**Pull request merged**